userspace/falco/app_actions/create_signal_handlers.cpp

```diff
@@ -74,21 +72,32 @@ bool application::create_handler(int sig, void (*func)(int), run_result &ret)
 
 application::run_result application::create_signal_handlers()
 {
+	falco::app::g_terminate.store(APP_SIGNAL_NOT_SET, std::memory_order_seq_cst);
+	falco::app::g_restart.store(APP_SIGNAL_NOT_SET, std::memory_order_seq_cst);
+	falco::app::g_reopen_outputs.store(APP_SIGNAL_NOT_SET, std::memory_order_seq_cst);
+
+	if (!g_terminate.is_lock_free()
+		|| !g_restart.is_lock_free()
+		|| !g_reopen_outputs.is_lock_free())
+	{
+		falco_logger::log(LOG_WARNING, "Bundled atomics implementation is not lock-free, signal handlers may be unstable\n");
+	}
+
 	run_result ret;
-	s_app = *this;
-	if(! create_handler(SIGINT, ::signal_callback, ret) ||
-	   ! create_handler(SIGTERM, ::signal_callback, ret) ||
-	   ! create_handler(SIGUSR1, ::reopen_outputs, ret) ||
-	   ! create_handler(SIGHUP, ::restart_falco, ret))
+	if(! create_handler(SIGINT, ::terminate_signal_handler, ret) ||
+	   ! create_handler(SIGTERM, ::terminate_signal_handler, ret) ||
+	   ! create_handler(SIGUSR1, ::reopen_outputs_signal_handler, ret) ||
+	   ! create_handler(SIGHUP, ::restart_signal_handler, ret))
```

Comment:
what about something like:
```suggestion
	return !create_handler(SIGINT, ::terminate_signal_handler, ret) ||
	   !create_handler(SIGTERM, ::terminate_signal_handler, ret) ||
	   ! create_handler(SIGUSR1, ::reopen_outputs_signal_handler, ret) ||
	   ! create_handler(SIGHUP, ::restart_signal_handler, ret)) ? run_result::fatal("") : run_result::ok();
```

Or if we don't want to return fatal just `run_result::ok()` in both cases :thinking: just a nitpick feel free to ignore

Reply:
`create_handler` already sets its own error results in the passed-in `ret`, so errors are already catched.